collect_app/src/main/java/org/odk/collect/android/preferences/dialogs/ResetDialogPreferenceFragmentCompat.kt

```diff
@@ -0,0 +1,284 @@
+package org.odk.collect.android.preferences.dialogs
+
+import android.content.Context
+import android.content.DialogInterface
+import android.graphics.Color
+import android.os.Bundle
+import android.view.View
+import android.widget.CompoundButton
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.widget.AppCompatCheckBox
+import androidx.preference.PreferenceDialogFragmentCompat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.odk.collect.android.R
+import org.odk.collect.android.activities.CollectAbstractActivity
+import org.odk.collect.android.fragments.dialogs.ResetSettingsResultDialog
+import org.odk.collect.android.injection.DaggerUtils
+import org.odk.collect.android.utilities.ProjectResetter
+import org.odk.collect.android.utilities.ProjectResetter.ResetAction.RESET_PREFERENCES
+import org.odk.collect.androidshared.ui.DialogFragmentUtils.dismissDialog
+import org.odk.collect.androidshared.ui.DialogFragmentUtils.showIfNotShowing
+import timber.log.Timber
+import javax.inject.Inject
+
+class ResetDialogPreferenceFragmentCompat :
```

Comment:
I like converting classes that you are editing to kotlin but could you do that in a separate commit? It's better to have the fix in one commit so that you can clearly see what was changed.

Reply:
Done. I will remember to split it up into two commits in future.